### PR TITLE
Maya: local variable 'arnold_standins' referenced before assignment - OP-5542

### PR DESCRIPTION
**Pull request by tokejepsen**
## Changelog Description
MayaLookAssigner erroring when MTOA is not loaded:

```
# Traceback (most recent call last):
#   File "\openpype\hosts\maya\tools\mayalookassigner\app.py", line 272, in on_process_selected
#     nodes = list(set(item["nodes"]).difference(arnold_standins))
# UnboundLocalError: local variable 'arnold_standins' referenced before assignment
```

## Testing notes:
1. Unload `mtoa.mll`.
2. Assign a Vray look.

### CODE REVIEW

File: openpype/hosts/maya/tools/mayalookassigner/app.py [hunk 1]
```diff
@@ -263,14 +263,14 @@ def on_process_selected(self):
                 for standin in arnold_standins:
                     if standin in nodes:
                         arnold_standin.assign_look(standin, subset_name)
+
```
[the remark this answers is not in the document]
Comment (tokejepsen) [reply openpype/hosts/maya/tools/mayalookassigner/app.py:266]:
```suggestion

```

**Review by BigRoy — APPROVED** [review]

**Review by LiborBatek — APPROVED** [review]
When ```Arnold``` not loaded ```Look Assigner``` works normally and user is able to assign ```Look``` for renderer like ```Vray```.

image

Note: During the ```Look Assign``` action  ```Arnold``` being forced to load, is it desirable? 

Here is a log from script editor:

```
select -r screwdriver_A_modelMain_01_:_GRP ;
Using existing QApplication..
// openpype.hosts.maya.api.lib : Using look for the first time ..
// openpype.pipeline.load.utils : Running 'LookLoader' on 'screwdriver_A'
// SubsetLoader : {}
updateRenderOverride;
evalDeferred "shaderBallRendererMenuUpdate";
// Warning: line 1: filePathEditor: Attribute 'aiStandIn.dso' and label 'Standin' have been saved already.
// Warning: line 1: filePathEditor: Attribute 'aiVolume.filename' and label 'VDB' have been saved already.
// Warning: line 1: About -ltVersion is deprecated
// Warning: line 1: filePathEditor: Attribute 'aiImage.filename' and label 'Image' have been saved already.
// Warning: line 1: filePathEditor: Attribute 'aiPhotometricLight.aiFilename' and label 'IES' have been saved already.
import arnold
import mtoa
import mtoa.cmds.registerArnoldRenderer;mtoa.cmds.registerArnoldRenderer.registerArnoldRenderer()
updateRenderOverride;
# File read in  3.2 seconds.
updateRendererUI;
addOneTabToGlobalsWindow("arnold", "Common", "createArnoldRendererCommonGlobalsTab");
addOneTabToGlobalsWindow("arnold", "Arnold Renderer", "createArnoldRendererGlobalsTab");
addOneTabToGlobalsWindow("arnold", "System", "createArnoldRendererSystemTab");
addOneTabToGlobalsWindow("arnold", "AOVs", "createArnoldAOVTab");
addOneTabToGlobalsWindow("arnold", "Diagnostics", "createArnoldRendererDiagnosticsTab");
// Warning: };
// Warning: "C:/Program Files/Autodesk/Maya2023/scripts/others/imageFormats.mel" line 65.2 : Global variable is already initialized; this occurrence is ignored.
select -cl  ;
Good bye
Using existing QApplication..
```

**Comment by BigRoy** [comment]
> Note: During the Look Assign action Arnold being forced to load even when assigning Vray Look... Is it fine/desirable ? I can imagine that keeping Arnold not loaded would be preffered in this particular case. Question is if we are able to detect the look is not arnold based one....

I have a feeling that this is potentially the case because the look you're loading 'requires' `mtoa` because it was loaded in the original scene that look file was generated in.

Likely the `.ma` file of that look has a line near the top of the file that says something like: `require "mtoa"` or some other form of mentioning `mtoa` with a require statement. Is that the case @LiborBatek ?

**Comment by LiborBatek** [comment]
> Likely the published v-ray look `.ma` file has a line near the top of the file that says something like: `require "mtoa"` or some other form of mentioning `mtoa` with a require statement. Is that the case @LiborBatek ?

Yep! I have removed that line from look .ma file and now it doesnt force load ```mtoa.mll``` when the Vray for Look needed only.

So all good!

File: openpype/hosts/maya/tools/mayalookassigner/app.py [hunk 2]
```diff
@@ -263,14 +263,13 @@ def on_process_selected(self):
                 for standin in arnold_standins:
                     if standin in nodes:
                         arnold_standin.assign_look(standin, subset_name)
+                nodes = list(set(item["nodes"]).difference(arnold_standins))
```
Comment (BigRoy) [inline openpype/hosts/maya/tools/mayalookassigner/app.py:266]: Ok, so this logic actually is wrong if both mtoa and vrayformaya are loaded. The discarded v-ray proxies at that point in time are not discarded anymore. As such, I believe this (and the line for vray proxies) should be:

```suggestion
                nodes = list(set(nodes).difference(arnold_standins))
```

Right?

Comment (tokejepsen) [reply openpype/hosts/maya/tools/mayalookassigner/app.py]: Yeah, sure we can do that.

**Review by BigRoy — APPROVED** [review]
Changes look good to me. 👍

**Review by antirotor — APPROVED** [review]

**Comment by tokejepsen** [comment]
@antirotor can this be merged?

**Pull request merged**